iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -36,6 +36,21 @@ internal class AmqpConnectionPool : IAmqpUnitManager
                 {
                     AmqpConnectionHolder[] amqpConnectionHolders = ResolveConnectionGroup(deviceIdentity);
                     amqpConnectionHolder = ResolveConnectionByHashing(amqpConnectionHolders, deviceIdentity);
+
+                    // For group sas token authenticated devices over a multiplexed connection, the TokenRefresher
+                    // of the first client connecting will be used for generating the group sas tokens
+                    // and will be associated with the connection itself.
+                    // For this reason, if the device identity of the client is not the one associated with the
+                    // connection, the associated TokenRefresher can be safely disposed.
+                    // Note - This does not cause any identity related issues since the group sas tokens are generated
+                    // against the hub host as the intended audience (without the "device Id").
+                    if (deviceIdentity.AuthenticationModel == AuthenticationModel.SasGrouped
```

Comment:
Is there no way to avoid creating these for mux scenarios instead of disposing the unnecessary ones? Looks like all we need is a 1SAS token for the connection irrespective of the devices in the mux.

Reply:
There isn't, no. The client is "set up" in a different layer (in internal client, before the whole concept of transport pipeline comes into picture), and this is where the token refresher is set up. This layer doesn't have any visibility about multiplexing, which is a transport-level construct.